checkCopyrightHeader

```diff
@@ -0,0 +1,271 @@
+    FIXED=0
+    head -5 "$FILE" | grep -E "$COPYRIGHT_NEEDLE" > /dev/null
+    if [ $? -eq 0 ]; then
+        sed -i -E "1,5 s/$COPYRIGHT_NEEDLE/$CORRECT_ATTRIBUTION/" "$FILE"
+        echo "=> Fixed incorrect attribution!"
+        FIXED=1
+    else
+        echo "=> FAILED: $COPYRIGHT_NEEDLE not found in $FILE"
+    fi
+    return $FIXED
+}
+
+fixMissingLicenseHeader() {
+    local FILE=$1
+    local INTENDED_ATTRIBUTIONS=$2
+    local FILE_EXTENSION=$(echo "$FILE" | awk -F'.' '{print $NF}')
+
+    if [ "$FILE_EXTENSION" == "java" ]; then
+        THIS_LICENSE=$JAVA_LICENSE
+    elif [ "$FILE_EXTENSION" == "jsp" ]; then
+        THIS_LICENSE=$JSP_LICENSE
+    else
+        echo "Unknown file extension: $FILE_EXTENSION"
+        return 0
+    fi
+
+    THIS_LICENSE=${THIS_LICENSE//###TOKEN###/$INTENDED_ATTRIBUTIONS}
+
+    TMPFILE=$(mktemp)
+    echo "$THIS_LICENSE" > "$TMPFILE"
+    cat "$FILE" >> "$TMPFILE"
+    mv "$TMPFILE" "$FILE"
+    echo "=> Fixed missing license header in $FILE"
+}
+
+processFiles(){
+    # Only git tracked files
+    git ls-tree -r main --name-only | grep '\.java$\|\.jsp$' | sort -u | while read line
+    do
+        #echo Processing "$line"
+        if ! [ -n "${line}" ] || ! [ -e "${line}" ] ||
+            [[ ${line} == *"/package-info.java" ]] || # No functional code
+            [[ ${line} == "xmppserver/src/main/java/org/dom4j/io/XMPPPacketReader.java" ]] || # Special license via DOM4J
+            [[ ${line} == "xmppserver/src/main/java/org/jivesoftware/util/FastDateFormat.java" ]]; # Special license via Disney //TODO: https://igniterealtime.atlassian.net/browse/OF-2754
+                then
+            continue
+        fi
+
+        MISSING_LICENSE_HEADER=0
+        MISSING_IGNITE_MENTION=0
+        INCORRECT_IGNITE_MENTION=0
+
+        # Check for license
+        grep "Licensed under the Apache License" "$line" > /dev/null
+        if [ $? -eq 1 ]; then
+            MISSING_LICENSE_HEADER=1
+            FILE_CREATION_DATE=$(git log --diff-filter=A --follow --format=%ad --date="format:%Y-%m-%d" -1 -- "$line")
+            echo "Missing license header in: $line (created $FILE_CREATION_DATE)"
+        fi
+
+        # Calculate what the Jive attribution should be
+        existedInTheJiveSoftwareEra
+        EXISTED_IN_THE_JIVE_SOFTWARE_ERA=$?
+
+        if [ $EXISTED_IN_THE_JIVE_SOFTWARE_ERA -eq 1 ]; then
+            JIVE_ATTRIBUTION=$(getCopyrightAttributionForJive)
+
+        # I'm not sure I'm happy fixing copyright headers for other folk.
+        # Jive already did this themselves in https://github.com/igniterealtime/Openfire/commit/ff0fa13dc3f67ca269da7bceef161e4a49663259
+        # So I'm going to leave this commented out for now.
+        #    grep "Copyright .* Jive Software" "$line" > /dev/null
+        #    if [ $? -eq 1 ]; then
+        #        MISSING_JIVE_MENTION=1
+        #        echo "Copyright header not found in: $line - needs to read: $JIVE_ATTRIBUTION"
+        #    fi
```

Comment:
This is the right decision, right?

Reply:
I'm not sure what the decision is in the first place.

Reply:
When I was fixing stuff up, I wrote fixes for Jive Software's copyright header too, then commented it out. I decided they took care of that to the extent that they wanted to way back when and we shouldn't touch it, regardless of what git history says.